src/pages/Inventory.page.js

```diff
@@ -29,4 +31,67 @@ export class InventoryPage extends BaseSwagLabPage {
     async addItemToCartById(id) {
         await this.addItemToCartBtns.nth(id).click();
     }
+
+    async addItemToCartByIndex(index) {
+        const itemDiv = this.inventoryItems.nth(index);
+        const addButton = itemDiv.locator('.btn_inventory').first();
+        await addButton.click();
+    }
+
+    async getItemInfoById(id) {
+        const name = await this.inventoryItemName.nth(id).textContent();
+        const price = parseFloat((await this.inventoryItemPrice.nth(id).textContent()).replace('$', ''));
+        const description = await this.page.locator('.inventory_item_desc').nth(id).textContent();
+        return { name, price, description };
+    }
+
+    async getItemInfoByIndex(index) {
+        const name = await this.inventoryItemName.nth(index).textContent();
+        const price = parseFloat((await this.inventoryItemPrice.nth(index).textContent()).replace('$', ''));
+        const description = await this.page.locator('.inventory_item_desc').nth(index).textContent();
+        return { name, price, description };
+    }
+
+    async getItemsList() {
+        const items = await this.inventoryItems.allTextContents();
+        return items;
+    }
+
+    async addRandomItemsToCart() {
+        const addedItems = [];
+
+        const itemCount = await this.inventoryItems.count();
+
+        const count = Math.floor(Math.random() * itemCount) + 1;
+
+        const randomIndexes = this.generateRandomIndexes(itemCount, count);
+
+        for (const index of randomIndexes) {
+            await this.addItemToCartByIndex(index);
+
+            try {
+                const itemInfo = await this.getItemInfoByIndex(index);
+                addedItems.push(itemInfo);
+            } catch (error) {
+                console.error(`Error ${index}: ${error.message}`);
+            }
+        }
+
+        return addedItems;
+    }
+
+    generateRandomIndexes(maxIndex, count) {
```

Comment:
and this method should be moved to utils. it is not related directly to the page